### PR TITLE
Teaches simple animals how to dodge.

### PR DESCRIPTION
**Pull request by AnturK**
It's no secret simple animals are pretty weak, so this tries to change it without padding their health.
Adds basic swerve when approaching the target, and sidestep when fighting.

### CODE REVIEW

**Comment by ma44** [comment]
https://github.com/tgstation/tgstation/issues/6322

**Comment by Iamgoofball** [comment]
we could also make them proc to dodge when moused over

**Comment by yoyobatty** [comment]
How is dodge proc actually called considering it relies off of a bunch of boolean statement that never become true outside of the dodge proc...

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 1]
```diff
@@ -368,6 +404,22 @@
...
+
+/mob/living/simple_animal/hostile/proc/dodge(moving_to,move_direction)
+	//Assuming we move towards the target we want to swerve toward them to get closer
+	var/cdir = turn(move_direction,45)
+	var/ccdir = turn(move_direction,-45)
+	dodging = FALSE
+	. = Move(get_step(loc,pick(cdir,ccdir)))
+	if(!.)//Can't dodge there so we just carry on
+		. =  Move(moving_to,move_direction)
+	dodging = TRUE
```
Comment (yoyobatty) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: This is the only time dodging becomes true is here

Comment (AnturK) [reply code/modules/mob/living/simple_animal/hostile/hostile.dm]: This only restores it. This PR did not set dodging to true on any type. (see latest commit)

**Review by SpaceManiac — DISMISSED** [review]
Code looks fine and testmerge didn't break anything AFAIK

**Comment by SpaceManiac** [comment]
Conflicted with the other PR

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**